src/main/java/com/teamabnormals/abnormals_core/core/events/CompatEvents.java

```diff
@@ -33,22 +34,20 @@ public static void onRightClickEntity(PlayerInteractEvent.EntityInteract event)
 		ItemStack stack = event.getItemStack();
 		if (target instanceof IAgeableEntity && stack.getItem() == Items.POISONOUS_POTATO && ACConfig.ValuesHolder.isPoisonPotatoCompatEnabled() && ModList.get().isLoaded("quark")) {
 			PlayerEntity player = event.getPlayer();
-			CompoundNBT persistantData = target.getPersistentData();
-			if (((IAgeableEntity) target).getGrowingAge() < 0 && !persistantData.getBoolean(POISON_TAG)) {
-				if (!event.getWorld().isRemote) {
-					if (target.world.rand.nextDouble() < ACConfig.ValuesHolder.poisonEffectChance()) {
-						target.playSound(SoundEvents.ENTITY_GENERIC_EAT, 0.5f, 0.25f);
-						persistantData.putBoolean(POISON_TAG, true);
-						if (ACConfig.ValuesHolder.shouldPoisonEntity()) {
-							((LivingEntity) target).addPotionEffect(new EffectInstance(Effects.POISON, 200));
-						}
-					} else {
-						target.playSound(SoundEvents.ENTITY_GENERIC_EAT, 0.5f, 0.5f + target.world.rand.nextFloat() / 2);
+			CompoundNBT persistentData = target.getPersistentData();
+			if (((IAgeableEntity) target).canAge(true) && !persistentData.getBoolean(POISON_TAG)) {
+				if (target.world.rand.nextDouble() < ACConfig.ValuesHolder.poisonEffectChance()) {
+					target.playSound(SoundEvents.ENTITY_GENERIC_EAT, 0.5f, 0.25f);
+					persistentData.putBoolean(POISON_TAG, true);
+					if (ACConfig.ValuesHolder.shouldPoisonEntity()) {
+						((LivingEntity) target).addPotionEffect(new EffectInstance(Effects.POISON, 200));
```

Comment:
This line here doesn't check that the entity is an instance of `LivingEntity`.
This is fine, but it should be stated on the doc of `IAgeableEntity` that implementations of it must also be an instance of `LivingEntity`.

Reply:
Latest commit should fix this.